Problems building Nemo 0.6.2 in OSX

I am having problems to build Nemo in OSX, both using Julia 0.5.2 or 0.6.0-rc2. 

The final output of `Pkg.build("Nemo")` is the following:
```julia
tmp-mul_1.s:144:no such instruction: `mulx 24(%rsi), %rbx,%rax'
make[2]: *** [mul_1.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -D__GMP_WITHIN_GMP -I.. -DOPERATION_preinv_mod_1 -m64 -O2 -march=corei7 -mtune=corei7-avx -c preinv_mod_1.c  -fno-common -DPIC -o .libs/preinv_mod_1.o
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```
I did not get any further by doing `julia build.jl` in `deps/`. I think the problem is related to mpir-3.0.0.

Any ideas or suggestions?

Did you do Pkg.rm("Nemo") first?

If that doesn't work, there's probably an issue with your machine not
supporting the new Intel instructions, which means it's being incorrectly
detected.

Can you go into .julia/v0.5/Nemo/deps/mpir-3.0.0 and run

./config.guess

and tell us the output, and then give us the output of

cat /proc/cpuinfo

or whatever the equivalent is on a Mac.

On 1 June 2017 at 00:54, Luis Benet <notifications@github.com> wrote:

> I am having problems to build Nemo in OSX, both using Julia 0.5.2 or
> 0.6.0-rc2.
>
> The final output of Pkg.build("Nemo") is the following:
>
> tmp-mul_1.s:144:no such instruction: `mulx 24(%rsi), %rbx,%rax'make[2]: *** [mul_1.lo] Error 1make[2]: *** Waiting for unfinished jobs....libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -D__GMP_WITHIN_GMP -I.. -DOPERATION_preinv_mod_1 -m64 -O2 -march=corei7 -mtune=corei7-avx -c preinv_mod_1.c  -fno-common -DPIC -o .libs/preinv_mod_1.omake[1]: *** [all-recursive] Error 1make: *** [all] Error 2
>
> I did not get any further by doing julia build.jl in deps/. I think the
> problem is related to mpir-3.0.0.
>
> Any ideas or suggestions?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Nemocas/Nemo.jl/issues/183>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOzpn9eU-7irNneszrYYYlJADuZiOiKks5r_e-YgaJpZM4NsS7c>
> .
>

Thanks for the reply. 

I just did the `Pkg.rm("Nemo")`, `Pkg.add("Nemo")` (and `Pkg.buil("Nemo")` in Julia 0.5, since in Julia0.6 seems to be automatic). In both cases it stops with an error message of the form:

```julia
hgcd2.c:38:no such instruction: `lzcnt %rdi, %rax'
hgcd2.c:38:no such instruction: `lzcnt %rdx, %rcx'
make[2]: *** [hgcd2.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

> Can you go into .julia/v0.5/Nemo/deps/mpir-3.0.0 and run `./config.guess`

haswell-apple-darwin13.4.0

> and tell us the output, and then give us the output of `cat /proc/cpuinfo` or whatever the equivalent is on a Mac.

$ sysctl -n machdep.cpu.brand_string
Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz

--- a/deps/build.jl
+++ b/deps/build.jl
@@ -4,6 +4,11 @@ pkgdir = dirname(dirname(@__FILE__))
 wdir = joinpath(pkgdir, "deps")
 vdir = joinpath(pkgdir, "local")
 
+if is_apple() && !("CC" in keys(ENV))
+   ENV["CC"] = "clang"
+   ENV["CXX"] = "clang++"
+end
+
 if !ispath(vdir)
 
     mkdir(vdir)

